app/User.php

```diff
@@ -374,4 +397,48 @@ public function accessOverrideBy(): BelongsTo
     {
         return $this->belongsTo(self::class, 'access_override_by_id');
     }
+
+    /**
+     * Synchronizes major relationship with a given list of gtAccountEntitlements.
+     *
+     * @param array<string>  $accountEntitlements
+     */
+    public function syncMajorsFromAccountEntitlements(array $accountEntitlements): void
+    {
+        $current_major_ids = $this->majors()->get()->map(static function (Major $item): int {
```

Comment:
The `map` call could be simplified to `pluck('id')` but fixing it isn't necessary.